service/interpreter/interfaces.go

```diff
@@ -103,6 +104,12 @@ type WorkflowProvider interface {
 	GetBackendType() service.BackendType
 	GetLogger(ctx UnifiedContext) UnifiedLogger
 	NewInterpreterContinueAsNewError(ctx UnifiedContext, input service.InterpreterWorkflowInput) error
+	NewSelector(ctx UnifiedContext) UnifiedSelector
```

Comment:
we don't have to use selector. In this case, just adding `ReceiveBlocking` to `ReceiveChannel`  will work better(simpler)